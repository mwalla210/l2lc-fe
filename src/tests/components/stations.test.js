import React from 'react'
import renderer from 'react-test-renderer'
import Stations from '../../components/stations'

jest.mock('../../components/barcode')
jest.mock('../../components/buttonDefault')
jest.mock('reactstrap', () => ({
  ButtonGroup: 'ButtonGroup',
  DropdownItem: 'DropdownItem',
  DropdownMenu: 'DropdownMenu',
  ButtonDropdown: 'ButtonDropdown',
  DropdownToggle: 'DropdownToggle',
}))

const defaultOptions = {
    printClick: jest.fn()
}

describe('Stations', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Stations {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders and calls function printClick', () => {
    let options = Object.assign({}, defaultOptions)
    global.print = jest.fn()
    const component = renderer.create(
      <Stations {...options}/>,
    )
    const inst = component.getInstance()
    inst.printClick()
    expect(global.print).toBeCalled()
  })
})
