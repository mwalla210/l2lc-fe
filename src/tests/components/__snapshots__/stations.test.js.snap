// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stations Renders with snapshot 1`] = `
<div>
  <div
    className="row justify-content-center"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      className="row justify-content-center"
    >
      <ButtonGroup />
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Receiving
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Ticketing
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Preparation
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Coating
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Big Blaster
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Decorative Coating
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Packaging
      </h1>
    </div>
    <div
      className="col-12"
    >
      <h1
        style={
          Object {
            "marginBottom": "3px",
            "marginTop": "50px",
          }
        }
      >
        Other
      </h1>
    </div>
  </div>
</div>
`;
